spec/services/queue_name_service_spec.rb

```diff
@@ -0,0 +1,32 @@
+require "rails_helper"
+
+RSpec.describe QueueNameService do
+  let(:use_case) { 'one' }
+
+  describe ".call" do
+    subject(:call) { described_class.call(use_case) }
+
+    context "when the service is called in UAT for use case one" do
+      before do
+        Rails.configuration.x.host_env = "uat"
+        Rails.configuration.x.status.app_branch = "this-is/a.test-branch"
+      end
+
+      it "prefixes the submission queue name with the branch name" do
+        expect(call).to eq "uc-one-this-is-a-test-branch-submissions"
+      end
+    end
+
+    context "when the service is called anywhere other than UAT for use case two" do
+      let(:use_case) { 'two' }
+
+      before do
+        Rails.configuration.x.host_env = "test"
```

Comment:
see [comment](https://github.com/ministryofjustice/laa-assure-hmrc-data/pull/177/files#r1190741462)

Reply:
👍 have amended but tweaked suggestion slightly as rubocop didn't like message_chain (this cop https://www.rubydoc.info/gems/rubocop-rspec/1.7.0/RuboCop/Cop/RSpec/MessageChain)